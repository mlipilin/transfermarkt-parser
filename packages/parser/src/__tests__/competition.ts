import { competition } from '../index'

const timeout = Number(process.env.JEST_ASYNC_TIMEOUT)

describe('list', () => {
  test(
    'competition.list(189) -> list of competitions for England',
    async () => {
      await expect(competition.list(189)).resolves.toMatchSnapshot()
    },
    timeout
  )
  test(
    'competition.list(-999) -> []',
    async () => {
      await expect(competition.list(-999)).resolves.toMatchSnapshot()
    },
    timeout
  )
  test(
    'competition.list() -> []',
    async () => {
      // @ts-ignore
      await expect(competition.list()).resolves.toMatchSnapshot()
    },
    timeout
  )
})
