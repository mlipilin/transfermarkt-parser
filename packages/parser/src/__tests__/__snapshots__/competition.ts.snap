// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list competition.list() -> [] 1`] = `[]`;

exports[`list competition.list(-999) -> [] 1`] = `[]`;

exports[`list competition.list(189) -> list of competitions for England 1`] = `
[
  {
    "countryId": 189,
    "id": "GB1",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/gb1.png",
    "title": "Premier League",
  },
  {
    "countryId": 189,
    "id": "GB2",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/gb2.png",
    "title": "Championship",
  },
  {
    "countryId": 189,
    "id": "GB3",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/gb3.png",
    "title": "League One",
  },
  {
    "countryId": 189,
    "id": "GB4",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/gb4.png",
    "title": "League Two",
  },
  {
    "countryId": 189,
    "id": "CNAT",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/cnat.png",
    "title": "National League",
  },
  {
    "countryId": 189,
    "id": "NLN6",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/nln6.png",
    "title": "National League North",
  },
  {
    "countryId": 189,
    "id": "NLS6",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/nls6.png",
    "title": "National League South",
  },
  {
    "countryId": 189,
    "id": "GB18",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/gb18.png",
    "title": "U18 Premier League",
  },
  {
    "countryId": 189,
    "id": "FAC",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/fac.png",
    "title": "FA Cup",
  },
  {
    "countryId": 189,
    "id": "GBCS",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/gbcs.png",
    "title": "Community Shield",
  },
  {
    "countryId": 189,
    "id": "CGB",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/cgb.png",
    "title": "EFL Cup",
  },
  {
    "countryId": 189,
    "id": "FAYC",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/fayc.png",
    "title": "FA Youth Cup",
  },
  {
    "countryId": 189,
    "id": "GB21",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/gb21.png",
    "title": "Premier League 2",
  },
  {
    "countryId": 189,
    "id": "GBFL",
    "logoUrl": "https://tmssl.akamaized.net//images/logo/normal/gbfl.png",
    "title": "EFL Trophy",
  },
]
`;
